kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -318,7 +319,9 @@ public static Map<String, String> validateDeltaProperties(Map<String, String> ne
       String key = kv.getKey().toLowerCase(Locale.ROOT);
       String value = kv.getValue();
 
-      if (key.startsWith("delta.")) {
+      // TableFeature properties validation is handled separately in TransactionBuilder.
```

Comment:
At this point, we should assign `key.startsWith(..) && !key.startsWith(...)` to a local variable with a good descriptive name, and then `if` on that. Let's make the entry requirements into this if statement very clear to readers.